.github/workflows/autograding.yml

```diff
@@ -20,54 +20,25 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version-file: 'pyproject.toml' # Read python version from a file pyproject.toml
+        cache: 'pip'
 
-    # Step 3: Install Poetry
-    - name: Install Poetry
+
+    # Check Task 2: Poetry Setup
+    - name: Check Task 2
       run: |
         python --version
         python -m pip install --upgrade pip
         pip install poetry
 
-    # Step 4: Install Dependencies
-    - name: Install Dependencies
-      run: |
         poetry install
         
-    # Check Task 2: Poetry Setup
-    - name: Check Task 2
-      run: |
         if [ -f "pyproject.toml" ] && [ -f "poetry.lock" ]; then
           echo "Poetry setup verified."
         else
           echo "Error: Poetry files not found."
           exit 1
         fi
```

Comment:
I think if these two files don't exist then `poetry install` might not work?

Reply:
Yep that's true, I'll refactor.